lms/static/scripts/frontend_apps/services/client-rpc.js

```diff
@@ -24,17 +43,44 @@ export class ClientRpc {
    * @param {Object} options
    *   @param {string[]} options.allowedOrigins -
    *     Origins that are allowed to request client configuration
-   *   @param {Object} options.clientConfig -
+   *   @param {string} options.authToken -
+   *     Auth token used in LMS backend API calls to refresh the grant token
+   *     in the `clientConfig` if it has expired
+   *   @param {ClientConfig} options.clientConfig -
    *     Configuration for the Hypothesis client. Whatever is provided here is
    *     passed directly to the client via `window.postMessage` when it requests
    *     configuration. It should be a subset of the config options specified at
    *     https://h.readthedocs.io/projects/client/en/latest/publishers/config/.
    */
-  constructor({ allowedOrigins, clientConfig }) {
+  constructor({ allowedOrigins, authToken, clientConfig }) {
     this._server = new Server(allowedOrigins);
 
-    // Handle the initial request for configuration from the Hypothesis client.
-    this._server.register('requestConfig', () => clientConfig);
+    // A convervative estimate of when the grant token was issued.
+    // When this is older than the true value, the frontend will just consider it
+    // to "expire" earlier than it really does.
+    const issuedAt = Date.now() - 30 * 1000;
+    let grantToken = new JWT(clientConfig.services[0].grantToken, issuedAt);
```

Comment:
This could raise an exception if the JWT is malformed. Maybe it could be wrap into a try/catch?

Reply:
If the JWT is malformed here then it means there is a programming error on the backend, in which case we just want a helpful exception so we can locate the problem. I've modified the `JWT` class so that it reports a more helpful error (eg. "Failed to parse JWT" rather than a generic "JSON parse error" or similar).